src/openms/source/DATASTRUCTURES/LPWrapper.cpp

```diff
@@ -66,23 +66,28 @@ namespace OpenMS
 {
 
 
-  LPWrapper::LPWrapper()
+  LPWrapper::LPWrapper(SOLVER solver)
```

Comment:
not a good idea.
All current code just calls a default c'tor.
This means that now, `solver` is set to COINOR (default-flag).
If COINOR is not installed, then the CTor will effectively do nothing and the object will be in an invalid state.

Currently, the `setSolver()` has a similar problem.

We need to decide if choosing between solvers is even desirable.
If COINOR is available, it should be used, because its much faster than GLPK for all scenarios I know of.

Thus the class choses the solver automatically and we should remove the `setSolver()` member to avoid trouble.

Reply:
> We need to decide if choosing between solvers is even desirable.
> If COINOR is available, it should be used, because its much faster than GLPK for all scenarios I know of.
> 
> Thus the class choses the solver automatically and we should remove the `setSolver()` member to avoid trouble.

I think this is the most straight forward solution.  An earlier solution that I tried that does just that is moving the GLPK sections below the `#if` COINOR section into an `#else` section so that by default COINOR is chosen if available and GLPK is used as a fall back.  However, I noticed many sections of the code and also tests are set up to run GLPK even if COINOR is available so I went with the current implementation as there would be quite a bit of refactoring needed on parts of the code that I am completely unfamiliar with.

Please let me know how you think we should move forward on this one.  I am not familiar with the other callers to LPWrapper well enough to make the call on this one in regards to what solver is most appropriate.  From my experience in optimization, I can say in general that COINOR is faster and and more accurate than GLPK, and in particular for our use cases COINOR would be preferred.  However, if there are application specific nuances in e.g., proteomics, I would not be in a good position to make that decision.

Reply:
I'd say that is enough anecdotal evidence :)
Lets just go ahead then and use COIN-OR if available and fall back to GLPK if needed.
Can you just revive the default Ctor and remove the setSolver(), and then lets see what happens.